pylmod/stellargradebook.py

```diff
@@ -0,0 +1,555 @@
+#!/usr/bin/python
+#
+# Python interface to Stellar Grade Book module
+#
+# Defines the class StellarGradeBook
+
+import csv
+import json
+import logging
+import time
+
+import requests
+
+
+log = logging.getLogger(__name__)
+
+
+class StellarGradeBook(object):
+    """
+    Python class representing interface to Stellar gradebook.
+
+    Example usage:
+
+    sg = StellarGradeBook('ichuang-cert.pem')
+    ats = sg.get('academicterms')
+    tc = ats['data'][0]['termCode']
+    sg.get('academicterm',termCode=tc)
+
+    students = sg.get_students()
+    assignments = sg.get_assignments()
+    sg.create_assignment('midterm1', 'mid1', 1.0, 100.0, '11-04-2013')
+
+    sid, student = sg.get_student_by_email(email)
+    aid, assignment = sg.get_assignment_by_name('midterm1')
+    sg.set_grade(aid, sid, 95.2)
+
+    sg.spreadsheet2gradebook(datafn)
+
+    """
+
+    URLBASE = 'https://learning-modules-test.mit.edu:8443/service/gradebook'
+
+    GETS = {'academicterms': '',
+            'academicterm': '/{termCode}',
+            'gradebook': '?uuid={uuid}',
+    }
+
+    GBUUID = 'STELLAR:/project/mitxdemosite'
+    TIMEOUT = 200  # connection timeout, seconds
+
+    verbose = True
+    gradebookid = None
+
+    # todo.do we want to pass gbuuid as param in __init__?
+    def __init__(self, cert, urlbase=None, gbuuid=None):
+        """
+        Initialize StellarGradeBook instance.
+
+          - urlbase:    URL base for gradebook API (defaults to self.URLBASE)
+            (still needs certs); default False
+          - gbuuid:     gradebook UUID (eg STELLAR:/project/mitxdemosite)
```

Comment:
You aren't adding this param to self here.  I think this is why I am getting 500's